pkg/permissions/verbs.go

```diff
@@ -18,36 +21,87 @@ const (
 	DELETE = Verb("DELETE")
 )
 
+var allVerbsOrder = []Verb{GET, POST, PUT, PATCH, DELETE}
+
 // VerbSet is a Set of Verbs
-type VerbSet []Verb
+type VerbSet map[Verb]bool
+
+// Contains check if VerbSet contains a Verb
+func (vs VerbSet) Contains(v Verb) bool {
+	if len(vs) == 0 {
+		return true // empty set = ALL
+	}
+	allowed, has := vs[v]
+	return has && allowed
+}
 
 func (vs VerbSet) String() string {
 	out := ""
 	if len(vs) == 0 || len(vs) == allVerbsLength {
 		return allVerbs
 	}
-	for _, v := range vs {
-		out += verbSep + string(v)
+	for _, v := range allVerbsOrder {
+		if vs[v] {
+			out += verbSep + string(v)
+		}
 	}
 	return out[1:]
 }
 
+// MarshalJSON implements json.Marshaller on VerbSet
+// the VerbSet is converted to a json array
+func (vs VerbSet) MarshalJSON() ([]byte, error) {
+	s := make([]string, len(vs))
+	i := 0
+	for _, v := range allVerbsOrder {
```

Comment:
You don't iter on `vs` directly to preserve the order of the verbs (GET always POST) ?